src/video/cocoa/cocoa_v.mm

```diff
@@ -836,31 +566,187 @@ static void ClearWindowBuffer(uint32 *buffer, uint32 pitch, uint32 height)
 	if (this->buffer_depth == 8) {
 		free(this->pixel_buffer);
 		this->pixel_buffer = malloc(this->window_width * this->window_height);
-		if (this->pixel_buffer == NULL) {
-			DEBUG(driver, 0, "Failed to allocate pixel buffer");
-			return false;
-		}
+		if (this->pixel_buffer == nullptr) usererror("Out of memory allocating pixel buffer");
+	} else {
+		free(this->pixel_buffer);
+		this->pixel_buffer = nullptr;
 	}
 
-	static_cast<VideoDriver_Cocoa *>(VideoDriver::GetInstance())->GameSizeChanged();
-
 	/* Redraw screen */
 	this->num_dirty_rects = MAX_DIRTY_RECTS;
+	this->GameSizeChanged();
+}
+
+void VideoDriver_Cocoa::CheckPaletteAnim()
+{
+	if (_cur_palette.count_dirty != 0) {
+		Blitter *blitter = BlitterFactory::GetCurrentBlitter();
+
+		switch (blitter->UsePaletteAnimation()) {
+			case Blitter::PALETTE_ANIMATION_VIDEO_BACKEND:
+				this->UpdatePalette(_cur_palette.first_dirty, _cur_palette.count_dirty);
+				break;
+
+			case Blitter::PALETTE_ANIMATION_BLITTER:
+				blitter->PaletteAnimate(_cur_palette);
+				break;
+
+			case Blitter::PALETTE_ANIMATION_NONE:
+				break;
+
+			default:
+				NOT_REACHED();
+		}
+		_cur_palette.count_dirty = 0;
+	}
+}
+
+
+bool VideoDriver_Cocoa::PollEvent()
+{
+#ifdef _DEBUG
+	uint32 et0 = GetTick();
+#endif
+	NSEvent *event = [ NSApp nextEventMatchingMask:NSAnyEventMask
+				untilDate:[ NSDate distantPast ]
+				inMode:NSDefaultRunLoopMode dequeue:YES ];
```

Comment:
Weird alignment going on here. 3 tabs? It's not that long, just stick it all on one line instead?